pkg/kube/secretcontroller/secretcontroller.go

```diff
@@ -39,8 +38,13 @@ const (
 	maxRetries = 5
 )
 
-// Required for unit test override.
-var loadKubeConfig = clientcmd.Load
+// LoadKubeConfig is a unit test override variable for loading the k8s config.
+// DO NOT USE - TEST ONLY.
+var LoadKubeConfig = clientcmd.Load
+
+// CreateInterfaceFromClusterConfig is a unit test override variable for interface create.
+// DO NOT USE - TEST ONLY.
+var CreateInterfaceFromClusterConfig = kube.CreateInterfaceFromClusterConfig
```

Comment:
I don't see that you need to define a variable here. You passed in a mocked clientset in line 433 in the test file

Reply:
Needed to mock out kube.CreateInterfaceFromClusterConfig for consumers of this library. Calling kube.CreateInterfaceFromClusterConfig without a 'real' clientConfig crashes the unit test.